### ISSUE TITLE
[Export]ed property doesn't narrow down possible object types to CustomResource type

### ISSUE THREAD

**Issue by xahon** [Atlinx/Godot-Mono-CustomResourceRegistry#11]
I've added a custom resource type and [Export]ed it in one of the scripts. When I'm assigning an object it shows me all possible resource types known to the engine

![image](https://user-images.githubusercontent.com/18731681/226312500-152d0531-44c2-4199-9acb-d302d570b9c1.png)

It's the same behaviour as I just make property type in a script as `[Export] Resource myCustomResource`

**Comment by Atlinx** [comment]
This is Godot's built in behaviour. This won't be fixed until we get first class support for C# resources.

See https://github.com/godotengine/godot/pull/72619.

In the mean time, you can assign a class prefix in the settings for this plugin. For example if you assign their prefix as _, all your custom resources will show up first.

**Comment by Nikola526** [comment]
Does it actually show up in the dropdown for you guys? I'm only seeing the custom resources in the file system's New Resource window.

**Comment by Atlinx** [comment]
> Does it actually show up in the dropdown for you guys? I'm only seeing the custom resources in the file system's New Resource window.

Uh oh you're right it's missing. We should probably file a bug report on the godot repo.

**Comment by Atlinx** [comment]
I've made the issue here https://github.com/godotengine/godot/issues/75245, although once the C# first class resource support gets added, this plugin will be moot.

**Comment by dashdotdashdot** [comment]
So until that issue is seen to or the C# first class resource support is finally merged (the former seems unlikely given the attention on the latter), our sole option for custom Resources written in C# is to create each instance individually through the filesystem's right click New > Resource menu and drag them into Inspector fields?

**Comment by Atlinx** [comment]
> So until that issue is seen to or the C# first class resource support is finally merged (the former seems unlikely given the attention on the latter), our sole option for custom Resources written in C# is to create each instance individually through the filesystem's right click New > Resource menu and drag them into Inspector fields?

Yeah I guess you can no longer create local custom resources in scenes. Creating resources in the file system should still work.

**Comment by dashdotdashdot** [comment]
> Yeah I guess you can no longer create local custom resources in scenes. Creating resources in the file system should still work.

I've found a workaround that permits local custom resources, even with their being absent from the new resource dropdown menu:

- Create the custom resource through the FileSystem tab.
- Drag it into a Resource field in the Inspector.
- Make the resource unique by right clicking it and selecting Make Unique.

This will copy the resource's contents into the .tscn file as though it had been created from the dropdown. From there, more local instances can be conveniently created by copying, pasting, and making the pasted resource unique.

Thank you for creating this plugin, by the way - it's an important stopgap while the main project works out C# first class resource support.

**Comment by Nikola526** [comment]
If you're dealing with an exported array of custom resources, you can also add the [Tool] attribute and then write a setter property for the array which auto initializes empty elements. This means that when you press the Add Element button, an object will be created automatically. This won't really work for child classes though, unless you write an inspector plugin.

**Comment by Alexcellent** [comment]
Hey maybe I'm late to the party, but you can add [GlobalClass] to narrow down the resources possibilities on [Export]

Godot 4.5.1 on Mac

<img width="708" height="81" alt="Image" src="https://github.com/user-attachments/assets/16abb3a6-eea5-41a9-91fd-9979c694ccb7" />
<img width="708" height="167" alt="Image" src="https://github.com/user-attachments/assets/1c6fe58c-fcea-4df6-b99c-48d08d408711" />
